docs/Migrations.rst

```diff
@@ -0,0 +1,260 @@
+Migrations
+==================
+
+Neo4j.rb provides an ``ActiveRecord``-like migration framework and a set of helper methods to manipulate both database schema and data.
+
+
+Generators
+------------------
+
+Migrations can be created by using the built-in Rails generator:
+
+.. code-block:: bash
+
+  rails generate neo4j:migration RenameUserNameToFirstName
+
+This will generate a new file located in ``db/neo4j/migrate/xxxxxxxxxx_rename_user_name_to_first_name.rb``
+
+.. code-block:: ruby
+  class RenameUserNameToFirstName < Neo4j::Migrations::Base
+    def up
+    end
+
+    def down
+    end
+  end
+
+In the same way as ``ActiveRecord`` does, you should fill up the ``up`` and ``down`` methods to define the migration and (eventually) the rollback steps.
+
+
+Transactions
+------------------
+Every migrations runs inside a transaction by default. So, if some statement fails inside a migration fails, the database rollbacks to the previous state.
+
+However this behaviour is not always good. For instance, neo4j doesn't allow schema and data changes in the same transaction.
+
+To disable this, you can use the ``disable_transactions!`` helper in your migration definition:
+
+.. code-block:: ruby
+
+  class SomeMigration < Neo4j::Migrations::Base
+    disable_transactions!
+
+    ...
+  end
+
+Neo4j::Migrations::Helpers
+------------------
+
+#execute
+~~~~~~
+
+Executes a pure neo4j cypher query, interpolating parameters.
+
+.. code-block:: ruby
+
+  execute('MATCH (n) WHERE n.name = {node_name} RETURN n', node_name: 'John')
+
+.. code-block:: ruby
+
+  execute('MATCH (n)-[r:`friend`]->() WHERE n.age = 7 DELETE r')
+
+
+#query
+---------------
+
+An alias for ``Neo4j::Session.query``. You can use it as root for the query builder:
+
+.. code-block:: ruby
+
+  query.match(:n).where(name: 'John').delete(:n)
+
+
+#remove_property
+---------------
+
+Removes a property given a label.
+
+:Ruby:
+  .. code-block:: ruby
+
+    remove_property(:User, :money)
+
+:Cypher:
+  .. code-block:: cypher
+
+    MATCH (n:`User`) REMOVE n.money
```

Comment:
In thinking about this I went back and forth, but I think we don't need the `Cypher`.  If we ever change how the helper executes then we'd need to remember to update the docs.

This makes me realize, though, that we **should** make sure we're displaying the `Cypher` which is being executed during a migration run.  Do you know if that's currently happening?  If not, it should be possible to just set the log level (see the `railtie.rb`)